docs/rgb_decomposition.md

```diff
@@ -0,0 +1,53 @@
+RGB Decomposition
+=================
+
+RGB products generated with HyP3 use both co-pol (VV or HH) and cross-pol (VH or HV) backscatter values to generate a color image from SAR data. This approach facilitates visual interpretation by decomposing the signals into surface scattering with some volume scattering (red band), volume scattering (green band), and surface scattering with very low volume scattering (blue band). 
+
+The dual-pol SAR dataset first undergoes radiometric terrain correction (RTC) to adjust for the distortions present in the images due to the side-looking geometry of SAR data acquisition. The RTC images are output in amplitude scale, and the RGB values are calculated using a combination of the VV and VH (or HH and HV) pixel values. The output RGB GeoTIFF is scaled so that the values for each band range from 1 to 255.
+
+### Decomposition Calculations
+
+Note that dual-pol image acquisition may provide either VV/VH returns or HH/HV returns, depending on the sensor mode. The same decomposition approach can be applied to either mode; refer to the appropriate equations below. 
+
+To calculate the RGB values, the data is first separated into surface (P<sub>r</sub>) and volume (P<sub>v</sub>) scattering components as follows:
+
+##### VV/VH
+
+P<sub>r</sub> = |S<sub>vv</sub>| <sup>2</sup> - 3 |S<sub>vh</sub>| <sup>2</sup>
+
+P<sub>v</sub> = 4 |S<sub>vh</sub>| <sup>2</sup>
+
+##### HH/HV
+
+P<sub>r</sub> = |S<sub>hh</sub>| <sup>2</sup> - 3 |S<sub>hv</sub>| <sup>2</sup>
+
+P<sub>v</sub> = 4 |S<sub>hv</sub>| <sup>2</sup>
+
+A spatial mask is generated using a threshold (*k*, typically between -22 and -25 dB) applied to the P<sub>v</sub> data so that pixels with values below the threshold are included in the blue channel mask (M<sub>B</sub>) and those above the threshold are included in the red band mask (M<sub>R</sub>).
+
+M<sub>B</sub> = P<sub>v</sub> < *k*
+
+M<sub>R</sub> = P<sub>v</sub> >= *k*
+
+These spatial masks are then applied to the P<sub>r</sub> data to assign the pixels to either the red or the blue band.
+
+P<sub>B</sub> = P<sub>r</sub> M<sub>B</sub>
```

Comment:
This should be negated:
```
P<sub>B</sub> = - P<sub>r</sub> M<sub>B</sub>
```

Reply:
Did that negation really happen in the code? I thought once the threshold was changed it worked out that you could have the mask as above (all pixels with values less than k in the cross-pol were used for blue and the pixels with values above k were used for red without any additional shenaniganry.

Reply:
Yeah, in the code, blue has a term written as:

```
3 * S<sub>XP</sub> -  S<sub>CP</sub>
```
which is 
```
3* S<sub>XP</sub> -  S<sub>CP</sub> = -1 * (S<sub>CP</sub> - 3 * S<sub>XP</sub>)  = -1 * P<sub>r</sub>
```
the mask is still used like you said.

Reply:
https://github.com/ASFHyP3/hyp3-lib/blob/develop/hyp3lib/rtc2color.py#L108

Reply:
The code you referenced took me to the line for teal. I thought you said we weren't actually applying the teal color option in the code; I expunged all references to it. Should we add that in?

Reply:
In the code: 
```python
    elif color == 'blue':
        z_constant = 5.0
        color_term = np.zeros(copol_data.shape)
    elif color == 'teal':
        z_constant = 5.0
        color_term = 2.0 * np.sqrt(np.clip(3.0 * crosspol_data - copol_data, 0, None))
```

For RTC browse, we always call it like: `rtc2color(ampfile, ampfile2, threshold, outfile, amp=True, cleanup=True)`
For RGB Decomp process, we always call it liike: `rtc2color(fullpol, crosspol, threshold, out_tiff, cleanup=True, teal=True)`

I think we should *not* mention teal in the docs, be write the equations like `teal=True` since that's how RGB Decomp is always run. We can revisit it if we decide to drop teal, or drop what is currently blue